lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -2185,12 +2186,16 @@ bool postProcessBuiltinReturningStruct(Function *F) {
                                      F->getReturnType());
       NewF->addParamAttr(0, SretAttr);
       NewF->setCallingConv(F->getCallingConv());
+      AllocaInst *A = Builder.CreateAlloca(F->getReturnType());
```

Comment:
But we should already have the appropriate alloca, haven't we? Like in the test case:
```
  %0 = alloca {i8, i8} // we should reuse it, otherwise generated code would be on the dead path
  call void @_Z17__spirv_IAddCarrycc(ptr sret ({i8, i8}) %0, i8 %a, i8 %b)
  ret void
```
we would need to create it only in a case, when we have no alloca in the IR before the translator, aka when you would try to translate the intrinsic.